frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx

```diff
@@ -151,9 +151,10 @@ const ImageSearch: React.FC = () => {
         helpText={getHelpText()}
         helpTextInvalid={helpTextInvalid}
         validated={validated}
-        onBlur={() => {
-          handleSearch();
-          setFieldTouched('searchTerm', true);
+        onChange={(e: KeyboardEvent) => {
+          setFieldValue('disableSubmit', true);
```

Comment:
I really don't think this field is necessary. We were able to have a disabled submit button without it. You should be resetting the values previously set by the error or success conditions. See what is done in the `catch` of the search.

Reply:
disabled submit button is based on only two props i.e. `dirty` and `errors`. Not sure how we can disable the button based on these and in catch block we are only setting the image stream data to empty values and error.

Reply:
The submit button disabled state is controlled by the validation of the form data. But deleting the required image values the form will be in an invalid state, thereby disabling the submit button.

By not having a valid image, the submit button will be disabled. So setting the values back to empty (as they were on initializing the form), we get the desired outcome.

Reply:
We can remove the `disableSubmit` prop from everywhere and  here we can set  `setFieldValue('isi', {});` so the form will be disabled if there is no valid image in the isi

Reply:
fixed!